src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupScheduleService.java

```diff
@@ -125,17 +125,8 @@ private TutorialGroupSession generateScheduledSession(Course course, TutorialGro
         session.setTutorialGroupSchedule(tutorialGroupSchedule);
         session.setTutorialGroup(tutorialGroupSchedule.getTutorialGroup());
 
-        var overlappingPeriod = tutorialGroupFreePeriodService.findOverlappingPeriod(course, session);
-        if (overlappingPeriod.isPresent()) {
-            session.setStatus(TutorialGroupSessionStatus.CANCELLED);
-            session.setStatusExplanation(null);
-            session.setTutorialGroupFreePeriod(overlappingPeriod.get());
-        }
-        else {
-            session.setStatus(TutorialGroupSessionStatus.ACTIVE);
-            session.setStatusExplanation(null);
-            session.setTutorialGroupFreePeriod(null);
-        }
+        var overlappingPeriod = tutorialGroupFreePeriodService.findOverlappingPeriod(course, session).stream().findFirst();
```

Comment:
Here again I'm unclear why we handle only the first overlap, aren't we interested in the remaining ones?

Reply:
We are only interested in the first one, as there can only be one freePeriod associated with a tutorialGroupSession. This is the case, as there would otherwise be a conflict in displaying the reason for a canceled session.